nixos/modules/services/web-apps/shiny-server.nix

```diff
@@ -0,0 +1,129 @@
+          extraRPackages = cfg.extraPackages;
+        };
+        defaultText = ''pkgs.shiny-server.override {
+          extraRPackages = cfg.extraPackages;
+        };
+        '';
+        description = "Shiny Server package";
+      };
+
+      extraPackages = mkOption {
+        default = [];
+        example = ''with pkgs.rPackages; [ ggplot2 data_table ];'';
+        description = ''
+          Extra packages to include in the rWrapper used by shiny-server.
+          Use this to enable additional R packages.
+        '';
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 3838;
+        description = ''
+          Port number Shiny Server will be listening to.
+        '';
+      };
+
+      directoryIndex = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether directory content is displayed";
+      };
+
+      user = mkOption {
+        default = "shiny";
+        description = "User to run Shiny Server as";
+      };
+
+      siteDir = mkOption {
+        type = types.path;
+        default = "${cfg.package}/shiny-server/samples";
+        defaultText = ''''${cfg.package}/shiny-server/samples'';
+        description = "Directory containing shiny apps to serve";
+      };
+      logDir = mkOption {
+        type = types.path;
+        default = "/var/lib/shiny/log";
+        description = "Directory to write server logs";
+      };
+      stateDir = mkOption {
+        type = types.path;
+        default = "/var/lib/shiny/state";
+        description = "Directory to store server state (bookmarks)";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Extra lines to be added to the generated configuration file.
+        '';
+      };
+
+      configFile = mkOption {
+        type = types.path;
+        default = generatedConfig;
+        defaultText = "generatedConfig";
+        description = ''
+          Overridable configuration file. Defaults is the one generated.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.extraUsers = optionalAttrs (cfg.user == "shiny") (singleton {
+      name = cfg.user;
+      uid = config.ids.uids.shiny;
+      group = cfg.user;
+      description = "Shiny Server user";
+    });
+    users.extraGroups = optionalAttrs (cfg.user == "shiny") (singleton {
+      name = cfg.user;
+      gid = config.ids.gids.shiny;
+    });
+
+    systemd.services.shiny-server = {
+      description = "Shiny Application Server";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      preStart = ''
+        mkdir -p ${cfg.stateDir} ${cfg.logDir}
+        chown ${cfg.user} ${cfg.stateDir} ${cfg.logDir}
+      '';
+
+      serviceConfig = {
+        ExecStart = "${pkgs.su}/bin/su -s ${pkgs.bash}/bin/sh ${cfg.user} -c '${cfg.package}/bin/shiny-server ${cfg.configFile}'";
```

Comment:
The `su` and `sh` bits can be replaced by setting:

```
        User = cfg.user;
        Group = cfg.group;
```

inside the serviceConfig.

Reply:
Done. Didn't know about PermissionsStartOnly.